enhancements/installer/component-selection.md

```diff
@@ -0,0 +1,522 @@
+be excluded).
+
+Specifying the same capability as an `include` and an `exclude` is an error condition.  This avoids the need 
+to define behavior for what to do when a user specifies the same capability in both, and update a status field
+indicating what choice the CVO made.  To enforce this the CVO will need to evaluate the spec values and, if
+they are valid, copy them into status.  If they are not valid, status must contain an indication of what is
+invalid and why.  This also means the CVO must be driven off the status values, not the spec values.
+
+Note: The CVO must wait until the ClusterVersion resource exists (created by the installer) before creating
+any filterable resource(a resource with a capability annotation) to ensure there is no race condition in which
+the CVO starts creating resources that should have been filtered, before it has the filter list.
+
+4) Admin can remove an item from the excluded annotations list, but they cannot add an item to it.  If an
+item is removed, the CVO will apply the previously filtered(skipped) resources to the cluster on the next reconciliation.
+Adding an item to the filtered list is not supported because it requires the component be removed from the
+running cluster which has more significant implications for how all traces of the component are removed.  The
+CVO should reject admin edits/updates that add new entries to the exclusion list.  Again this requires the
+introduction of a validating webhook for the ClusterVersion resource.
+
+Similarly, changing the `inclusionDefault` from `include` to `exclude` must be prevented.
+
+The currently configured capabilities settings for the CVO should be recorded in telemeter so we can understand
+the configuration of a given cluster.
+
+In the future, we might allow specific APIs to be disabled, such as the build api.  This could be done by
+defining additional capability keywords that can be put into the install-config field being defined here,
+which would drive the creation of cluster config that the apiserver+controllers used to disable that particular
+api.
+
+5) During upgrade, resources may move from being associated with one capability, to another, or move
+from being part of the core(always installed) to being part of an optional capability.  In such cases,
+the governing logic is that once a resource has been applied to a cluster, it must continue to be
+applied to the cluster(unless it is removed from the payload entirely, of course).  Furthermore,
+if any resource from a given capability is applied to a cluster, all other resources from that
+capability must be applied to the cluster, even if the cluster configuration would otherwise filter
+out that capability.  These are referred to as implicitly enabled capabilities and the purpose is to
+avoid scenarios like this that would break cluster functionality:
+
+1) initially registry resources are part of core(not defined in a capability)
```

Comment:
The rendering is not working properly here. This is showing at the same level as the previous point instead of a sub section of bullet points.
![Screenshot from 2022-01-12 18-03-29](https://user-images.githubusercontent.com/3125299/149237047-b30b8368-f29f-441a-9dc8-f17f36674974.png)

Reply:
fixed (somewhat hackily)